### PR TITLE
chore: adding type alias for code hash byte slices

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #4993 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/keeper/genesis.go [hunk 1]
```diff
@@ -29,7 +31,7 @@ func (k Keeper) ExportGenesis(ctx sdk.Context) types.GenesisState {
 	// Grab code from wasmVM and add to genesis state.
 	var genesisState types.GenesisState
 	for _, codeHash := range codeHashes {
-		code, err := k.wasmVM.GetCode(codeHash)
+		code, err := k.wasmVM.GetCode(wasmvmtypes.Checksum(codeHash))
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/keeper/genesis.go:34]: Adding the type alias means we need to do some type conversions when calling into the vm.

The `Checksum` type in `wasmvm` is also just an alias on `[]byte`

File: modules/light-clients/08-wasm/types/wasm.go [hunk 2]
```diff
@@ -6,29 +6,33 @@ import (
...
 	if err != nil {
 		return nil, err
 	}
 
-	if codeHashes == nil {
-		codeHashes = [][]byte{}
+	var codeHashes []CodeHash
+	for _, key := range keys {
+		codeHashes = append(codeHashes, key)
 	}
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/types/wasm.go:28]: I removed this check as I figured `nil` should be a valid return value. From what I gathered the only place needing this was in `genesis_test.go` [here](https://github.com/cosmos/ibc-go/pull/5004/files#diff-5a718389573601b8fa391f2362feabf88a431f525e6adc2971c6f1f3f60b2619L62).

**Review by chatton — APPROVED** [review]
LGTM 👍

File: modules/light-clients/08-wasm/types/wasm.go (hunk 2 above)
Comment (DimitrisJim) [reply modules/light-clients/08-wasm/types/wasm.go:28]: It's an either or I guess but in general not having a possibly `nil` value floating around is a slight preference. not a hill I'm dying on though

**Review by DimitrisJim — APPROVED** [review]

File: modules/light-clients/08-wasm/types/wasm.go [hunk 3]
```diff
@@ -6,29 +6,33 @@ import (
...
 
-	codeHashes, err := iterator.Keys()
+	keys, err := iterator.Keys()
 	if err != nil {
 		return nil, err
 	}
 
-	if codeHashes == nil {
-		codeHashes = [][]byte{}
+	codeHashes := []CodeHash{}
```
Comment (DimitrisJim) [inline modules/light-clients/08-wasm/types/wasm.go:25]: sweet alt

**Review by colin-axner — APPROVED** [review]

**Pull request merged**